transactor/src/main/scala/io/mediachain/copycat/TransactorService.scala

```diff
@@ -129,17 +129,24 @@ extends ClientStateListener with JournalListener {
   }
   
   private def journalCommitEvent(entry: JournalEntry) {
-    if (state.streaming) {
+    if (state.streaming && entry.index >= state.index) {
+      if (entry.index != state.index) {
+        logger.warn(s"Commit skipped entries ${state.index} -> ${entry.index}")
+      }
       state.block += entry
-      state.index = entry.index
+      state.index = entry.index + 1
       emitEvent(TransactorService.journalEntryToEvent(entry)) 
     }
   }
   
-  private def journalBlockEvent(ref: Reference) {
-    if (state.streaming) {
+  private def journalBlockEvent(ref: Reference, index: BigInt) {
+    if (state.streaming && index >= state.index) {
```

Comment:
so the index is checked in 2 different places, but we're guaranteed no race between these, right? can these two events happen concurrently?